stdarch: update submodule

KisaragiEffective opened the pull request:
We need [this commit](https://github.com/rust-lang/stdarch/commit/cf3deeae3a420fc8b409b596aeda6ce7027067d3) introduced by [stdarch#1411](https://github.com/rust-lang/stdarch/pull/1411) in order to merge #110189.

Note to myself: `git pull && git submodule update --remote library/stdarch`

KisaragiEffective committed:
stdarch: update submodule

JohnTitor commented:
r? @Amanieu

Amanieu commented:
@bors r+

bors commented:
:pushpin: Commit d5357c1b399410b2559d32b2fbd4afa92c57d7fa has been approved by `Amanieu`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=Amanieu d5357c1b399410b2559d32b2fbd4afa92c57d7fa -->
<!-- homu: {"type":"Approved","sha":"d5357c1b399410b2559d32b2fbd4afa92c57d7fa","approver":"Amanieu","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

bors commented:
:hourglass: Testing commit d5357c1b399410b2559d32b2fbd4afa92c57d7fa with merge 727db2d74e04267f48bbe758ac55ac05168bf4c3...
<!-- homu: {"type":"BuildStarted","head_sha":"d5357c1b399410b2559d32b2fbd4afa92c57d7fa","merge_sha":"727db2d74e04267f48bbe758ac55ac05168bf4c3"} -->

bors commented:
:broken_heart: Test failed - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/4709180450/jobs/8352169145)
<!-- homu: {"type":"BuildFailed","builder_url":"https://github.com/rust-lang-ci/rust/actions/runs/4709180450/jobs/8352169145","builder_name":"checks-actions"} -->

rust-log-analyzer commented:
The job **`dist-powerpc64-linux`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/4709180450/jobs/8352083360) [(plain)](https://github.com/rust-lang-ci/rust/commit/727db2d74e04267f48bbe758ac55ac05168bf4c3/checks/12773802434/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
[RUSTC-TIMING] build_script_build test:false 0.463
error[E0412]: cannot find type `vector_double` in this scope
 --> library/core/src/../../portable-simd/crates/core_simd/src/vendor/powerpc.rs:9:35
  |
9 | from_transmute! { unsafe f64x2 => vector_double }

error[E0412]: cannot find type `vector_signed_long` in this scope
  --> library/core/src/../../portable-simd/crates/core_simd/src/vendor/powerpc.rs:10:35
   |
   |
10 | from_transmute! { unsafe i64x2 => vector_signed_long }
   |                                   ^^^^^^^^^^^^^^^^^^ help: a struct with a similar name exists: `vector_signed_int`
  ::: library/core/src/../../stdarch/crates/core_arch/src/macros.rs:62:9
   |
   |
62 |         pub struct $name($($fields)*);
   |         ------------------------------ similarly named struct `vector_signed_int` defined here

error[E0412]: cannot find type `vector_unsigned_long` in this scope
  --> library/core/src/../../portable-simd/crates/core_simd/src/vendor/powerpc.rs:11:35
   |
11 | from_transmute! { unsafe u64x2 => vector_unsigned_long }
   |                                   ^^^^^^^^^^^^^^^^^^^^ help: a struct with a similar name exists: `vector_unsigned_int`
  ::: library/core/src/../../stdarch/crates/core_arch/src/macros.rs:62:9
   |
   |
62 |         pub struct $name($($fields)*);
   |         ------------------------------ similarly named struct `vector_unsigned_int` defined here
error: unused import: `core::arch::powerpc64::*`
 --> library/core/src/../../portable-simd/crates/core_simd/src/vendor/powerpc.rs:7:5
  |
7 | use core::arch::powerpc64::*;
7 | use core::arch::powerpc64::*;
  |     ^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: `-D unused-imports` implied by `-D warnings`
For more information about this error, try `rustc --explain E0412`.
[RUSTC-TIMING] core test:false 6.352
error: could not compile `core` due to 4 previous errors
warning: build failed, waiting for other jobs to finish...
```

</details>

KisaragiEffective commented:
@bors retry

bors commented:
@KisaragiEffective: :key: Insufficient privileges: not in try users

Amanieu commented:
cc @lu-zero I think this failure may be due to your powerpc changes.

Amanieu commented:
This should be fixed by https://github.com/rust-lang/stdarch/pull/1420.

lu-zero commented:
Somehow the notification did not hit my mailbox properly, sorry for the noise.

KisaragiEffective committed:
stdarch: update submodule, take 2

Amanieu commented:
@bors r+

bors commented:
:pushpin: Commit 49413bb619e3ed8d589e7cf7f2d81127b7f6ce36 has been approved by `Amanieu`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=Amanieu 49413bb619e3ed8d589e7cf7f2d81127b7f6ce36 -->
<!-- homu: {"type":"Approved","sha":"49413bb619e3ed8d589e7cf7f2d81127b7f6ce36","approver":"Amanieu","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

matthiaskrgr commented:
@bors r-
https://github.com/rust-lang/rust/pull/110753#issuecomment-1519820049

KisaragiEffective committed:
stdarch: update submodule, take 3

Amanieu commented:
@bors r+

bors commented:
:pushpin: Commit 1042b65997084f3c9de06f1c5e64ace28cbe7a26 has been approved by `Amanieu`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=Amanieu 1042b65997084f3c9de06f1c5e64ace28cbe7a26 -->
<!-- homu: {"type":"Approved","sha":"1042b65997084f3c9de06f1c5e64ace28cbe7a26","approver":"Amanieu","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

bors commented:
:hourglass: Testing commit 1042b65997084f3c9de06f1c5e64ace28cbe7a26 with merge faca88b6c23897bb1e9053125de3518858ed3837...
<!-- homu: {"type":"BuildStarted","head_sha":"1042b65997084f3c9de06f1c5e64ace28cbe7a26","merge_sha":"faca88b6c23897bb1e9053125de3518858ed3837"} -->

rust-log-analyzer commented:
The job **`arm-android`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/4793648117/jobs/8526288585) [(plain)](https://github.com/rust-lang-ci/rust/commit/faca88b6c23897bb1e9053125de3518858ed3837/checks/12995450467/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
   Compiling std v0.0.0 (/checkout/library/std)
error: unknown arm target feature: crypto
  --> /checkout/library/stdarch/crates/std_detect/src/detect/macros.rs:62:17
   |
31 | /          macro_rules! $macro_name {
33 | |                  ($feature_lit) => {
33 | |                  ($feature_lit) => {
34 | |                      $crate::detect_feature!($feature, $feature_lit $(: $($target_feature_lit),*)?)
62 | |/                 compile_error!(
63 | ||                     concat!(
63 | ||                     concat!(
64 | ||                         concat!("unknown ", stringify!($target)),
65 | ||                         concat!(" target feature: ", $t)
67 | ||                 )
   | ||_________________^
68 | |              };
69 | |          }
69 | |          }
   | |__________- in this expansion of `is_arm_feature_detected!`
  ::: library/std/tests/run-time-detect.rs:19:28
   |
   |
19 |        println!("crypto: {}", is_arm_feature_detected!("crypto"));

[RUSTC-TIMING] run_time_detect test:true 0.046
error: could not compile `std` due to previous error
warning: build failed, waiting for other jobs to finish...
```

</details>

bors commented:
:broken_heart: Test failed - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/4793648117/jobs/8526606906)
<!-- homu: {"type":"BuildFailed","builder_url":"https://github.com/rust-lang-ci/rust/actions/runs/4793648117/jobs/8526606906","builder_name":"checks-actions"} -->

taiki-e commented:
The failure is related to https://github.com/rust-lang/stdarch/pull/1407.

Removing the following line should fix it.

https://github.com/rust-lang/rust/blob/f6830a212973ced91a047cee86cb79cee4f23868/library/std/tests/run-time-detect.rs#L19

KisaragiEffective committed:
stdarch: update submodule, take 4

Amanieu commented:
You need to apply the change in https://github.com/rust-lang/rust/pull/110285#issuecomment-1521201953 to your PR otherwise the tests will not pass.

KisaragiEffective committed:
stdarch: update submodule, take 5

KisaragiEffective commented:
It should be updated properly.

Amanieu commented:
No, you need to delete this line on the rust-lang/rust side: https://github.com/rust-lang/rust/blob/master/library/std/tests/run-time-detect.rs#L19

KisaragiEffective commented:
Oh, okay...

KisaragiEffective committed:
std: remove test for arm's crypto feature
please see https://github.com/rust-lang/rust/pull/110285#issuecomment-1521201953 for more details

Amanieu commented:
@bors r+

bors commented:
:pushpin: Commit a4014f08a77f5e1df79e04492440eb1dbdc478bc has been approved by `Amanieu`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=Amanieu a4014f08a77f5e1df79e04492440eb1dbdc478bc -->
<!-- homu: {"type":"Approved","sha":"a4014f08a77f5e1df79e04492440eb1dbdc478bc","approver":"Amanieu","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

bors commented:
:hourglass: Testing commit a4014f08a77f5e1df79e04492440eb1dbdc478bc with merge f7b831ac8a897273f78b9f47165cf8e54066ce4b...
<!-- homu: {"type":"BuildStarted","head_sha":"a4014f08a77f5e1df79e04492440eb1dbdc478bc","merge_sha":"f7b831ac8a897273f78b9f47165cf8e54066ce4b"} -->

bors commented:
:sunny: Test successful - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/4925663381/jobs/8804412629)
Approved by: Amanieu
Pushing f7b831ac8a897273f78b9f47165cf8e54066ce4b to master...
<!-- homu: {"type":"BuildCompleted","approved_by":"Amanieu","base_ref":"master","builders":{"checks-actions":"https://github.com/rust-lang-ci/rust/actions/runs/4925663381/jobs/8804412629"},"merge_sha":"f7b831ac8a897273f78b9f47165cf8e54066ce4b"} -->

rust-timer commented:
Finished benchmarking commit (f7b831ac8a897273f78b9f47165cf8e54066ce4b): [comparison URL](https://perf.rust-lang.org/compare.html?start=ecd3dbab4ed82abfa05e22069261e565239449cf&end=f7b831ac8a897273f78b9f47165cf8e54066ce4b&stat=instructions:u).

### Overall result: no relevant changes - no action needed


@rustbot label: -perf-regression



### [Instruction count](https://perf.rust-lang.org/compare.html?start=ecd3dbab4ed82abfa05e22069261e565239449cf&end=f7b831ac8a897273f78b9f47165cf8e54066ce4b&stat=instructions:u)
This benchmark run did not return any relevant results for this metric.

### [Max RSS (memory usage)](https://perf.rust-lang.org/compare.html?start=ecd3dbab4ed82abfa05e22069261e565239449cf&end=f7b831ac8a897273f78b9f47165cf8e54066ce4b&stat=max-rss)
<details>
<summary>Results</summary>

This is a less reliable metric that may be of interest but was not used to determine the overall result at the top of this comment.

|                                    | mean  | range          | count |
|:----------------------------------:|:-----:|:--------------:|:-----:|
| Regressions ❌ <br /> (primary)    | -     | -              | 0     |
| Regressions ❌ <br /> (secondary)  | 2.6%  | [2.0%, 3.0%]   | 3     |
| Improvements ✅ <br /> (primary)   | -1.0% | [-1.0%, -1.0%] | 1     |
| Improvements ✅ <br /> (secondary) | -     | -              | 0     |
| All ❌✅ (primary)                 | -1.0% | [-1.0%, -1.0%] | 1     |
</details>

### [Cycles](https://perf.rust-lang.org/compare.html?start=ecd3dbab4ed82abfa05e22069261e565239449cf&end=f7b831ac8a897273f78b9f47165cf8e54066ce4b&stat=cycles:u)
This benchmark run did not return any relevant results for this metric.

### [Binary size](https://perf.rust-lang.org/compare.html?start=ecd3dbab4ed82abfa05e22069261e565239449cf&end=f7b831ac8a897273f78b9f47165cf8e54066ce4b&stat=size:linked_artifact)
This benchmark run did not return any relevant results for this metric.

**Bootstrap**: 659.045s -> 658.646s (-0.06%)
